Add optional close button

cdwort opened the pull request:
We wanted to have tooltips be click to close (to accommodate clicking links inside) and found our users (really other devs roped into testing) had trouble finding how to close the tooltip. They first clicked outside the tooltip or inside the tooltip. Only after did they clicked on the original span they moused over/clicked on.

We wanted to add a close icon to provide a clear indication about where to click.

The resulting code to build up the template is little ugly, and this might be an edge use case, so no worries if this isn't something you're looking for.

Let me know if you'd like any changes or if I missed some documentation.

Thanks!

45kb commented:
@cdwort Hi, first of all thanks for the PR, this feature is great.

I did not had any chance to try it, but as from the code you pushed, i just have some doubt,
wouldn't be better to call close-button="" attribute something like tooltip-close-button="" ?

Also could be great to use plain html inside the button, for example when using fontawesome:

    tooltip-close-button="<i class='fa fa-remove'></i>"

If you can fix or this is already fixed, let know, so we'll be glad to merge the PR!

cdwort commented:
Hi @45kb, thanks for the fast response! 

Thanks for catching the wrong attribute naming. It's now called tooltip-close-button.

It already does accept HTML, and there is an example in the Readme. I added examples to the index.html page and added some default styles to move the button to the right.

Let me know if there's anything else you would like to see. I can squash commits if you would like.

wouldgo commented:
Hi @cdwort,
Looks good to me merging, could you please write down a brief explaination also in the readme.md file?

Best regard,
Dario.

45kb commented:
@cdwort  thanks for the help! if you can please finish it, as @wouldgo  suggested :+1: 

thank you once more.

cdwort commented:
Hi @wouldgo 

I'm happy to make changes to what I already added for the readme, but could you give me a little more detail about what you would like to see? Are you asking that I break it out into a new section? Since it's only relevant when you click to close, I thought it might make sense to live in the triggers section.

Thanks for feedback,
Amy

wouldgo commented:
@cdwort: ops! my fault I didn't saw your changes on the readme.md... Really sorry about that.
Merging right now, and thanks a lot for your commitment!

Dario.

cdwort commented:
@wouldgo @45kb I realize that for us to use this version we would probably want a minimized version. Would you accept another PR to update the minimized files?

45kb commented:
@cdwort  i just pushed the minified versions, soon we will update tag and bower too

cdwort commented:
Awesome thanks!

45kb commented:
@cdwort it's a pleasure , thanks for your time.
